src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -168,6 +170,49 @@ private IEnumerable<CompletionItem> GetResourceTypeCompletions(SemanticModel mod
                 .ToList();
         }
 
+        private IEnumerable<CompletionItem> GetModulePathCompletions(SemanticModel model, BicepCompletionContext context)
+        {
+            if (!context.Kind.HasFlag(BicepCompletionContextKind.ModulePath) 
+            || Path.GetDirectoryName(model.SyntaxTree.FileUri.LocalPath) is not string cwd
+            || context.EnclosingDeclaration is not ModuleDeclarationSyntax declarationSyntax
+            || declarationSyntax.Path is not StringSyntax stringSyntax
+            || stringSyntax.TryGetLiteralValue() is not string entered
+            || File.Exists(Path.Combine(cwd, entered)))
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+            
+            var query = new Uri(Path.Combine(cwd, entered));
```

Comment:
>entered [](start = 50, length = 7)

Does Path.Combine throw when `entered` is an absolute path?

Reply:
Also, I'm pretty sure it will throw when `entered` contains invalid path characters.

---
In reply to: [551557314](https://github.com/Azure/bicep/pull/1236#discussion_r551557314) [](ancestors = 551557314)

Reply:
I'll check this out. It is worth noting that our VScode extension (and probably cli as well) does not allow absolute module paths

Reply:
Yes, but nothing blocks the user from typing in something horribly malformed. If we consume that and it throws, it will crash the language server.